app/services/discourse_reactions/reaction_like_synchronizer.rb

```diff
@@ -0,0 +1,439 @@
+        ON CONFLICT DO NOTHING
+      SQL
+      inserted_user_action_count =
+        DB.exec(sql_query, ua_was_liked: UserAction::WAS_LIKED, post_action_ids: post_action_ids)
+      Rails.logger.info(
+        "[ReactionsLikeSynchronizer] Inserted #{inserted_user_action_count} was_liked UserActions",
+      )
+    end
+
+    # Delete any UserAction records for LIKE or WAS_LIKED that match up with
+    # PostAction records that got trashed.
+    def self.delete_excluded_user_actions(trashed_post_action_ids)
+      return if trashed_post_action_ids.empty?
+
+      sql_query = <<~SQL
+        DELETE FROM user_actions
+        WHERE id IN (
+          -- Select IDs for LIKED actions
+          SELECT user_actions.id
+          FROM user_actions
+          INNER JOIN post_actions ON user_actions.target_post_id = post_actions.post_id
+            AND user_actions.acting_user_id = post_actions.user_id
+          WHERE post_actions.id IN (:trashed_post_action_ids)
+          AND user_actions.action_type = :ua_like
+
+          UNION
+
+          -- Select IDs for WAS_LIKED actions
+          SELECT user_actions.id
+          FROM user_actions user_actions
+          INNER JOIN post_actions ON user_actions.target_post_id = post_actions.post_id
+          INNER JOIN posts ON posts.id = post_actions.post_id
+          WHERE post_actions.id IN (:trashed_post_action_ids)
+          AND user_actions.action_type = :ua_was_liked
+          AND user_actions.user_id = posts.user_id
+          AND user_actions.acting_user_id = post_actions.user_id
+        )
+      SQL
+      DB.exec(
+        sql_query,
+        ua_like: UserAction::LIKE,
+        ua_was_liked: UserAction::WAS_LIKED,
+        trashed_post_action_ids: trashed_post_action_ids,
+      )
+    end
+
+    # Find all PostAction records that have a ReactionUser record that
+    # uses a reaction in the excluded_from_like list and trash them.
+    def self.trash_excluded_post_actions(excluded_from_like)
+      return [] if excluded_from_like.none?
+
+      sql_query = <<~SQL
+        WITH deleted_post_actions AS (
+          UPDATE post_actions
+          SET deleted_at = NOW()
+          FROM discourse_reactions_reaction_users ru
+          INNER JOIN discourse_reactions_reactions ON discourse_reactions_reactions.id = ru.reaction_id
+          WHERE post_actions.user_id = ru.user_id
+            AND post_actions.post_id = ru.post_id
+            AND post_actions.post_action_type_id = :like
+            AND discourse_reactions_reactions.reaction_value IN (:excluded_from_like)
+          RETURNING post_actions.id, post_actions.post_id, post_actions.user_id
+        )
+
+        DELETE FROM user_actions
+        USING deleted_post_actions
+        WHERE user_actions.target_post_id = deleted_post_actions.post_id
+        AND user_actions.acting_user_id = deleted_post_actions.user_id
+        AND user_actions.action_type IN (:ua_like, :ua_was_liked)
```

Comment:
You are deleting user actions here so I think that method `delete_excluded_user_actions` is not needed anymore. Or keep `delete_excluded_user_actions` and remove block above.

Reply:
Oh nice catch, I will delete this block